src/main/java/org/opensrp/web/rest/EventResource.java

```diff
@@ -537,6 +558,114 @@ public Event update(Event entity) {
 			return new ResponseEntity<>(INTERNAL_SERVER_ERROR);
 		}
 	}
+
+	@GetMapping(value = "/export-data", produces = "application/zip")
+	public ResponseEntity<ByteArrayResource> exportEventData(@RequestParam List<String> eventTypes,
+			@RequestParam String planIdentifier,
+			@RequestParam(value = "fromDate", required = false) String fromDate,
+			@RequestParam(value = "toDate", required = false) String toDate) throws IOException {
+
+		FileOutputStream fos = null;
+		ZipOutputStream zipOS = null;
+		String exportDataFileName;
+		String missionName = "";
+		String eventTypeName;
+		SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+		String formatted = "";
+		String zipFileName = "";
+		boolean firstTime = true;
+
+		for (String eventType : eventTypes) {
+			ExportEventDataSummary exportEventDataSummary = eventService
+					.exportEventData(planIdentifier, eventType, Utils.getDateTimeFromString(fromDate),
+							Utils.getDateTimeFromString(toDate));
+
+			missionName = exportEventDataSummary != null &&
+					exportEventDataSummary.getMissionName() != null ?
+					exportEventDataSummary.getMissionName().replaceAll("\\s+", "_").toLowerCase() :
+					"";
+			eventTypeName = eventType.replaceAll("\\s+", "_");
+
+			formatted = df.format(new Date());
+			File csvFile = null;
+			File imagesDirectory = null;
+			zipFileName = SAMPLE_CSV_FILE + missionName + "_" + formatted + ".zip";
+			if (firstTime) {
+				fos = new FileOutputStream(zipFileName);
+				zipOS = new ZipOutputStream(fos);
+			}
+			try {

```

Comment:
use try with resources so that resource are closed

Reply:
Resources are closed outside the loop.
The resources needs to be opened for one time only since for all event types a single zip output stream will be used.

Reply:
Moreover, we can not take try block outside the loop, since we will be getting the missionName after executing the service method inside the loop which will be used in a zip file name. Based on the zip file name `zipOutputStream` will be opened.

Reply:
in that case case resources are usually closed in final block to make sure that they are still closed in case errors do occur

Reply:
Resources need to be closed outside a for loop.